core/shared/src/main/scala/org/typelevel/log4cats/noop/NoOpLoggerKernel.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.log4cats.noop
+
+import cats.Applicative
+import org.typelevel.log4cats.{KernelLogLevel, Log, LoggerKernel}
+
+class NoOpLoggerKernel[F[_]: Applicative, Ctx] extends LoggerKernel[F, Ctx] {
+  def log(level: KernelLogLevel, record: Log.Builder[Ctx] => Log.Builder[Ctx]): F[Unit] =
+    Applicative[F].unit
+}
+
+object NoOpLoggerKernel {
+  def apply[F[_]: Applicative, Ctx]: NoOpLoggerKernel[F, Ctx] = new NoOpLoggerKernel[F, Ctx]
+}
```

Comment:
No need for a class, it's both simpler (and a good test of the SAM-ness of `LoggerKernel`) to implement it directly
```suggestion
object NoOpLoggerKernel {
  def apply[F[_]: Applicative, Ctx]: LoggerKernel[F, Ctx] = new LoggerKernel[F, Ctx] {
    override def log(level: KernelLogLevel, record: Builder => Builder): F[Unit] =
      Applicative[F].unit
  }
}
```

Interestingly, this does highlight that `LoggerKernel` is not currently a SAM type